CMakeLists.txt

```diff
@@ -24,7 +24,9 @@ endif()
 
 if (ENABLE_CPP)
   enable_language(CXX)
-  set(CMAKE_CXX_STANDARD 11 CACHE STRING "C++ standard")
+  set(CMAKE_CXX_STANDARD 14 CACHE STRING "C++ standard")
```

Comment:
Can you please explain why we had to change this? Also shouldn't we do this in case of autotools as well?

Reply:
`CMAKE_CXX_STANDARD_REQUIRED` makes sure we don't require C++14 forcefully.
It's far from perfect, but I think allowing the usage of `make_unique()` is worth it.